Set/Set/SetGame.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  SetGame.swift
+//  Set
+//
+//  Created by Yeliena Khaletska on 11.06.2024.
+//
+
+import Foundation
+
+struct Card {
+
+    let content = "💩"
+
+}
+
+final class SetGame {
+
+    private var deck: Array<Card>
+    private var shownCards: Array<Card>
+    private var chosenCardsIndices: Array<Int>
+    private var matchedCardsIndices: Array<Int>
+    var canDrawCards: Bool {
+        self.shownCards.count < 24
+    }
+
+    init() {
+        self.deck = Array(repeating: Card(), count: 81)
+        self.shownCards = []
+        self.chosenCardsIndices = []
+        self.matchedCardsIndices = []
+    }
+
+    func drawCards(_ amount: Int) {
+        for _ in 0..<amount {
+            drawOneCard()
+        }
+    }
+
+    private func drawOneCard() {
+        guard let lastCard = self.deck.popLast() else {
+            assertionFailure("not implemented")
+            return
+        }
+
+        self.shownCards.append(lastCard)
+    }
+
+    func touchCard(index: Int) {
+        if self.chosenCardsIndices.contains(index) {
+            chosenCardsIndices.removeAll { $0 == index }
+        }
+        else {
+            self.chosenCardsIndices.append(index)
+            if self.chosenCardsIndices.count == 3 {
+                if matchCards() {
+                    self.matchedCardsIndices.append(contentsOf: self.chosenCardsIndices)
+                    self.shownCards.remove(atOffsets: IndexSet(self.chosenCardsIndices))
+                    self.chosenCardsIndices = []
+                    drawCards(3)
+                }
+            }
+        }
```

Comment:
does it make sense for you to split 2 actions here. First we update card selection, then we try to match it. For example:
```suggestion
    updateSelectedState(for: card)
    
    if self.chosenCardsIndices.count == 3, matchCards() {
                markChosenCardsAsMatched()
                drawCards(3)
    }
```

Reply:
I won't add markChosenCardsAsMatched() for now because I need update shown cards as well and empty chosen cards so function name does not reflect what's happening inside. Will think and maybe create something with the next commit

Reply:
Okay?